### PR TITLE
refactor: (GenAI) Reorganized Token Count Samples (Group C)

### PR DESCRIPTION
**Pull request by Thoughtseize1**
## Description

Created new folder: `/generative_ai/token_count`

| Region_tag                                               | New file              |
|----------------------------------------------------------|-----------------------|
| generativeaionvertexai_token_count_sample_with_local_sdk | local_sdk_example.py  |
| generativeaionvertexai_token_count_sample_with_genai     | api_example.py        |
| generativeaionvertexai_gemini_token_count                | simple_example.py     |
| generativeaionvertexai_gemini_token_count_multimodal     | **multimodal_token_count_example.py** |

I renamed `multimodal_example.py` -> **multimodal_token_count_example.py** to prevent conflicts.

**❗Files need to be deleted after approval:**

- Folder - `/generative_ai/count_token`
- `generative_ai/gemini_count_token_example.py`
- [test_gemini_count_token_example()](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/302a01cefced5764cf271a40443bb0173d3f29f3/generative_ai/test_gemini_examples.py#L86-L93) from `generative_ai/test_gemini_examples.py`

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/token_count/simple_example.py
Comment (holtskinner) [inline generative_ai/token_count/simple_example.py:1]: Do we need this example in addition to the `api_example.py`? They seem to accomplish the same task

**Review by holtskinner — CHANGES_REQUESTED** [review]

File: generative_ai/token_count/multimodal_token_count_example.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+def count_tokens_multimodal() -> GenerationResponse:
+    # [START generativeaionvertexai_gemini_token_count_multimodal]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO(developer): Update and un-comment below line
+    # PROJECT_ID = "your-project-id"
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    model = GenerativeModel("gemini-1.5-flash-001")
```
Comment (msampathkumar) [inline generative_ai/token_count/multimodal_token_count_example.py:30]:
```suggestion
    model = GenerativeModel("gemini-1.5-flash-002")
```

File: generative_ai/token_count/simple_example.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+def count_tokens() -> GenerationResponse:
+    # [START generativeaionvertexai_gemini_token_count]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel
+
+    # TODO(developer): Update and un-comment below line
+    # PROJECT_ID = "your-project-id"
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    model = GenerativeModel("gemini-1.5-flash-001")
```
Comment (msampathkumar) [inline generative_ai/token_count/simple_example.py:30]:
```suggestion
    model = GenerativeModel("gemini-1.5-flash-002")
```

File: generative_ai/token_count/simple_example.py
Comment (msampathkumar) [reply generative_ai/token_count/simple_example.py:1]: Once the sample are segregated into individual folders, it might easier to find and fix duplicates.

**Review by msampathkumar — APPROVED** [review]
LGTM

**Comment by msampathkumar** [comment]
Holt's query is valid, but its outside the scope of this reorganization effort. The CICD for GenAI samples is taking too long due to duplicated code, so we're prioritizing a fast-paced approach to this reorganization.

**Pull request merged**